modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/schema/IgniteTableImpl.java

```diff
@@ -539,20 +540,33 @@ private class StatisticsImpl implements Statistic {
         /** {@inheritDoc} */
         @Override
         public Double getRowCount() {
-            if (statReqCnt.getAndIncrement() % STATS_CLI_UPDATE_THRESHOLD == 0) {
+            int prevCnt = statReqCnt.getAndIncrement();
+
+            if (prevCnt % STATS_CLI_UPDATE_THRESHOLD == 0) {
                 int parts = table.storage().distributionZoneConfiguration().partitions().value();
 
                 long size = 0L;
+                boolean noCache = false;
 
                 for (int p = 0; p < parts; ++p) {
                     @Nullable MvPartitionStorage part = table.storage().getMvPartition(p);
 
-                    if (part != null) {
+                    if (part == null) {
+                        continue;
+                    }
+
+                    try {
                         size += part.rowsCount();
+                    } catch (StorageRebalanceException ignore) {
+                        noCache = true;
                     }
                 }
 
-                localRowCnt = size;
+                if (noCache) {
```

Comment:
in case the first calculation is fallen we will use zero as the rows counter, I'm not sure but maybe will be better to use incorrectly calculated value for the case

Reply:
Agree, fixed a bit.

Reply:
I decided to revert this change completely... currently any concurrent call can see a zero in localRowCnt (if it is calculated for the first time), but may be it's ok.